scapy/utils.py

```diff
@@ -1918,13 +2106,30 @@ def write_header(self, pkt):
                     pkt.__class__
                 ]
             except KeyError:
-                warning("PcapWriter: unknown LL type for %s. Using type 1 (Ethernet)", pkt.__class__.__name__)  # noqa: E501
+                msg = "%s: unknown LL type for %s. Using type 1 (Ethernet)"
+                warning(msg, self.__class__, pkt.__class__.__name__)
                 self.linktype = DLT_EN10MB
         self._write_header(pkt)
 
+    def _get_time(self,
+                  packet,  # type: Union[bytes, Packet]
+                  sec,  # type: Optional[float]
+                  usec  # type: Optional[int]
+                  ):
+        # type: (...) -> Tuple[float, int]
+        if hasattr(packet, "time"):
```

Comment:
couldn't this be replaced with `isinstance(packet Packet)` ? or not isinstance bytes

Reply:
That makes sense. The typing will be cleaner.